zdx/mem.go

```diff
@@ -1,59 +1,109 @@
 package zdx
 
 import (
-	"errors"
+	"fmt"
 	"sort"
+
+	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
 )
 
-// MemTable implements a simple stub for an in-memory SST Reader.  This is
-// useful for testing and reference.  An SST client will typically implement its
-// own in-memory data structure that provides a Reader implementation to convert
-// custom key/value pairs into string/byte-slice pairs.
+// MemTable implements an in-memory table to build a zdx.
+// It implements zbuf.Reader and will generate a stream of zng.Records that
+// are either single column ("key") or a two-column ("key", "value") where the
+// types of the columns depend upon the zng.Values entered into the table.
 type MemTable struct {
-	table  map[string][]byte
-	keys   []string
-	offset int
+	table   map[string]zcode.Bytes
+	keys    []string
```

Comment:
why string and not []byte ?

Reply:
byte slices cannot be keys in the hash table so when you pull the keys back out they are already strings.  Converting to byte slices would create unnecessary copies.  That said, think things are still escaping to GC in the inner loop but we can fix this later with "unsafe" etc.

Reply:
I missed your point!  Sorting doesn't work right and we need to do comparisons of zng.Values.  Will fix.  We should address the GC escapes at some point.  I think it's ok to make the pointers all unsafe while the sort and write happens.